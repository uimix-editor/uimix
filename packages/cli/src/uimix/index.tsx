import React from "react";
import "./index.css";

function applyOverrides(
  rootProps: any,
  element: React.ReactElement
): React.ReactElement {
  const { "data-refID": refID, ...props } = element.props;

  let additionalProps: any = rootProps;
  for (const id of refID ?? []) {
    additionalProps = additionalProps[id + "Props"] ?? {};
  }
  additionalProps = { ...additionalProps };
  for (const key of Object.keys(additionalProps)) {
    if (key.endsWith("Props")) {
      delete additionalProps[key];
    }
  }

  const result = React.cloneElement(
    { ...element, props },
    {
      ...additionalProps,
      className: additionalProps.className
        ? props.className + " " + additionalProps.className
        : props.className,
      children:
        additionalProps.children ??
        React.Children.map(element.props.children, (child) => {
          if (React.isValidElement(child)) {
            return applyOverrides(rootProps, child);
          } else {
            return child;
          }
        }),
    }
  );
  return result;
}
