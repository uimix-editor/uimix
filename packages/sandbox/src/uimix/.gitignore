/components.css
/components.tsx
/images
