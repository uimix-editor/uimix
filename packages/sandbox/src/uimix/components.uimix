{
  "nodes": {
    "1Bvi2z3USBZi1pWrvPd6Ut": {
      "type": "component",
      "name": "Button",
      "index": 11
    },
    "1Bvi2z3UPfi6W3L8nDt5VA": {
      "type": "frame",
      "name": "",
      "parent": "1Bvi2z3USBZi1pWrvPd6Ut",
      "index": 0
    },
    "1Bvi2z3UGTM64JtRT7Qc1S": {
      "type": "text",
      "name": "Text",
      "parent": "1Bvi2z3UPfi6W3L8nDt5VA",
      "index": 0
    },
    "1Bvi2z3UVsEskiSt2TcdL7": {
      "type": "variant",
      "name": "",
      "condition": { "type": "hover" },
      "parent": "1Bvi2z3USBZi1pWrvPd6Ut",
      "index": 1
    },
    "1Bvi2z3UaHFmaikKtfcmGX": {
      "type": "variant",
      "name": "",
      "condition": { "type": "maxWidth", "value": 767 },
      "parent": "1Bvi2z3USBZi1pWrvPd6Ut",
      "index": 2
    },
    "1Bvi2zJALza4DrqmNaGYAk": {
      "type": "component",
      "name": "Stack",
      "index": 13
    },
    "1Bvi2z3TfkSpYXW29daTN8": {
      "type": "frame",
      "name": "Stack",
      "parent": "1Bvi2zJALza4DrqmNaGYAk",
      "index": 0
    },
    "1Bvi2z3TpKXtTJLTRPTEd8": {
      "type": "frame",
      "name": "Item 0",
      "parent": "1Bvi2z3TfkSpYXW29daTN8",
      "index": 0
    },
    "1Bvi2z3To8ddXo6R6ZftVD": {
      "type": "frame",
      "name": "Item 1",
      "parent": "1Bvi2z3TfkSpYXW29daTN8",
      "index": 1
    },
    "1BwspVKcGYvxVctYVzjPu6": {
      "type": "frame",
      "name": "Hidden Item",
      "parent": "1Bvi2z3TfkSpYXW29daTN8",
      "index": 1.5
    },
    "1Bvi2z3UB1xUVg4Xz9WTpJ": {
      "type": "frame",
      "name": "Item 2",
      "parent": "1Bvi2z3TfkSpYXW29daTN8",
      "index": 2
    },
    "1Bvi2z3U3SiK6r7gf9vh7N": {
      "type": "text",
      "name": "Text",
      "parent": "1Bvi2z3TfkSpYXW29daTN8",
      "index": 3
    },
    "1Bvi3BSSb2oaHb1ZqUEk6H": {
      "type": "component",
      "name": "Page",
      "index": 14
    },
    "1Bvi319YVxZELz4tEskN1S": {
      "type": "frame",
      "name": "Frame",
      "parent": "1Bvi3BSSb2oaHb1ZqUEk6H",
      "index": 0
    },
    "1Bvi31VKu8gfy6XzJ8xrHA": {
      "type": "foreign",
      "name": "Header",
      "parent": "1Bvi319YVxZELz4tEskN1S",
      "index": 1
    },
    "1Bvi365mtWZmXKDPeJTB4s": {
      "type": "frame",
      "name": "Frame",
      "parent": "1Bvi319YVxZELz4tEskN1S",
      "index": 3
    },
    "1Bvi32d9RMi3ES5kk2UvuM": {
      "type": "foreign",
      "name": "Button",
      "parent": "1Bvi365mtWZmXKDPeJTB4s",
      "index": 0
    },
    "1Bvi2z3UrU9VkawwAteXs8": {
      "type": "instance",
      "name": "Instance",
      "parent": "1Bvi365mtWZmXKDPeJTB4s",
      "index": 1
    },
    "1Bvi392qBdMy93LGLcVymb": {
      "type": "image",
      "name": "Image",
      "parent": "1Bvi365mtWZmXKDPeJTB4s",
      "index": 2
    },
    "1Bvi39mct9UNYZ9eTCCRdV": {
      "type": "instance",
      "name": "Stack",
      "parent": "1Bvi365mtWZmXKDPeJTB4s",
      "index": 3
    },
    "1BwWRiQqGA4gTdwZRLXFBh": {
      "type": "component",
      "name": "Card",
      "index": 15
    },
    "1BwWRgjHrhVbbuTV7m36kM": {
      "type": "frame",
      "name": "Frame 14",
      "parent": "1BwWRiQqGA4gTdwZRLXFBh",
      "index": 0
    },
    "1BwWRgjHyEzJjsehRefVBE": {
      "type": "frame",
      "name": "icon",
      "parent": "1BwWRgjHrhVbbuTV7m36kM",
      "index": 0
    },
    "1BwWRgjHr8HNgX8BRooGv7": {
      "type": "svg",
      "name": "heroicons:code-bracket",
      "parent": "1BwWRgjHyEzJjsehRefVBE",
      "index": 0
    },
    "1BwWRgjHmYZ8qLKDj3Tvdy": {
      "type": "text",
      "name": "title",
      "parent": "1BwWRgjHrhVbbuTV7m36kM",
      "index": 1
    },
    "1BwWRgjHuAe9UTxVhowwYu": {
      "type": "text",
      "name": "description",
      "parent": "1BwWRgjHrhVbbuTV7m36kM",
      "index": 2
    }
  },
  "styles": {
    "1Bvi2z3USBZi1pWrvPd6Ut": {},
    "1Bvi2z3UPfi6W3L8nDt5VA": {
      "position": {
        "x": { "type": "start", "start": 50 },
        "y": { "type": "start", "start": 400 }
      },
      "width": { "type": "hug" },
      "height": { "type": "hug" },
      "fills": [{ "type": "solid", "color": "#E5E7EB" }],
      "layout": "flex",
      "paddingTop": 4,
      "paddingRight": 8,
      "paddingBottom": 4,
      "paddingLeft": 8
    },
    "1Bvi2z3UGTM64JtRT7Qc1S": {
      "width": { "type": "hug" },
      "height": { "type": "hug" },
      "fills": [{ "type": "solid", "color": "#111827" }],
      "textContent": "Button"
    },
    "1Bvi2z3UVsEskiSt2TcdL7": {
      "position": {
        "x": { "type": "start", "start": 200 },
        "y": { "type": "start", "start": 400 }
      },
      "fills": [{ "type": "solid", "color": "#3B82F6" }]
    },
    "1Bvi2z3UVsEskiSt2TcdL7:1Bvi2z3UGTM64JtRT7Qc1S": {
      "fills": [{ "type": "solid", "color": "#FFFFFF" }]
    },
    "1Bvi2z3UaHFmaikKtfcmGX": {
      "position": {
        "x": { "type": "start", "start": 350 },
        "y": { "type": "start", "start": 400 }
      },
      "paddingTop": 2,
      "paddingRight": 4,
      "paddingBottom": 2,
      "paddingLeft": 4
    },
    "1Bvi2z3UaHFmaikKtfcmGX:1Bvi2z3UGTM64JtRT7Qc1S": { "fontSize": 14 },
    "1Bvi2zJALza4DrqmNaGYAk": {},
    "1Bvi2z3TfkSpYXW29daTN8": {
      "position": {
        "x": { "type": "start", "start": 50 },
        "y": { "type": "start", "start": 184 }
      },
      "width": { "type": "hug" },
      "height": { "type": "hug" },
      "fills": [{ "type": "solid", "color": "#FFFFFF" }],
      "layout": "flex",
      "rowGap": 10,
      "columnGap": 10,
      "paddingTop": 10,
      "paddingRight": 20,
      "paddingBottom": 30,
      "paddingLeft": 40
    },
    "1Bvi2z3TpKXtTJLTRPTEd8": {
      "width": { "type": "fill", "min": 25, "max": 100, "value": 50 },
      "height": { "type": "fill", "min": 20, "max": 60, "value": 50 },
      "fills": [
        {
          "type": "solid",
          "color": { "type": "token", "id": "1BwuQkMEqq1sH8PAPnDWRJ" }
        }
      ]
    },
    "1Bvi2z3To8ddXo6R6ZftVD": {
      "width": { "type": "fixed", "value": 40 },
      "height": { "type": "fixed", "value": 80 },
      "fills": [{ "type": "solid", "color": "#22C55E" }]
    },
    "1Bvi2z3UB1xUVg4Xz9WTpJ": {
      "width": { "type": "fixed", "value": 80 },
      "height": { "type": "fixed", "value": 40 },
      "fills": [{ "type": "solid", "color": "#3B82F6" }]
    },
    "1Bvi2z3U3SiK6r7gf9vh7N": {
      "width": { "type": "hug" },
      "height": { "type": "hug" },
      "fills": [{ "type": "solid", "color": "#000000" }],
      "marginTop": 10,
      "marginRight": 20,
      "marginBottom": 30,
      "marginLeft": 40,
      "textContent": "Hello, world!",
      "fontSize": 20
    },
    "1Bvi3BSSb2oaHb1ZqUEk6H": {},
    "1Bvi319YVxZELz4tEskN1S": {
      "position": {
        "x": { "type": "start", "start": 50 },
        "y": { "type": "start", "start": 585 }
      },
      "width": { "type": "fixed", "value": 744 },
      "height": { "type": "fixed", "value": 456 },
      "fills": [{ "type": "solid", "color": "#FFFFFF" }],
      "layout": "flex",
      "flexDirection": "y",
      "flexAlign": "center"
    },
    "1Bvi31VKu8gfy6XzJ8xrHA": {
      "width": { "type": "fill", "value": 744 },
      "height": { "type": "hug" },
      "foreignComponent": {
        "type": "react",
        "path": "src/stories/Header.tsx",
        "name": "Header",
        "props": {}
      }
    },
    "1Bvi365mtWZmXKDPeJTB4s": {
      "position": {
        "x": { "type": "start", "start": 210 },
        "y": { "type": "start", "start": 142 }
      },
      "width": { "type": "fill", "value": 428 },
      "height": { "type": "fill", "value": 173 },
      "fills": [{ "type": "solid", "color": "#FFFFFF" }]
    },
    "1Bvi32d9RMi3ES5kk2UvuM": {
      "position": {
        "x": { "type": "start", "start": 33 },
        "y": { "type": "start", "start": 28 }
      },
      "width": { "type": "hug" },
      "height": { "type": "hug" },
      "foreignComponent": {
        "type": "react",
        "path": "src/stories/Button.tsx",
        "name": "Button",
        "props": {
          "label": "Button1",
          "primary": true,
          "backgroundColor": "red",
          "size": "large"
        }
      }
    },
    "1Bvi2z3UrU9VkawwAteXs8": {
      "position": {
        "x": { "type": "start", "start": 160 },
        "y": { "type": "start", "start": 28 }
      },
      "fills": [{ "type": "solid", "color": "#BABABA" }],
      "mainComponent": "1Bvi2z3USBZi1pWrvPd6Ut"
    },
    "1Bvi2z3UrU9VkawwAteXs8:1Bvi2z3UGTM64JtRT7Qc1S": {
      "textContent": "Override Name"
    },
    "1Bvi392qBdMy93LGLcVymb": {
      "position": {
        "x": { "type": "start", "start": 33 },
        "y": { "type": "start", "start": 97 }
      },
      "width": { "type": "fixed", "value": 100 },
      "height": { "type": "fixed", "value": 100 },
      "fills": [{ "type": "solid", "color": "#FFFFFF" }],
      "imageHash": "YmY-tuteU03QY-k_2ARy1uacNmXDNmOX-FFAlj2qCIo="
    },
    "1Bvi39mct9UNYZ9eTCCRdV": {
      "position": {
        "x": { "type": "start", "start": 165 },
        "y": { "type": "start", "start": 86 }
      },
      "mainComponent": "1Bvi2zJALza4DrqmNaGYAk"
    },
    "1Bvi39mct9UNYZ9eTCCRdV:1Bvi2z3TpKXtTJLTRPTEd8": {},
    "1Bvi39mct9UNYZ9eTCCRdV:1Bvi2z3To8ddXo6R6ZftVD": {},
    "1Bvi39mct9UNYZ9eTCCRdV:1Bvi2z3UB1xUVg4Xz9WTpJ": {},
    "1Bvi39mct9UNYZ9eTCCRdV:1Bvi2z3U3SiK6r7gf9vh7N": {},
    "1BwWRgjHrhVbbuTV7m36kM": {
      "hidden": false,
      "position": {
        "x": { "type": "start", "start": 50 },
        "y": { "type": "start", "start": 1144 }
      },
      "width": { "type": "fixed", "value": 400 },
      "height": { "type": "hug" },
      "topLeftRadius": 20,
      "topRightRadius": 20,
      "bottomRightRadius": 20,
      "bottomLeftRadius": 20,
      "fills": [{ "type": "solid", "color": "#ffffff" }],
      "opacity": 1,
      "overflowHidden": false,
      "layout": "flex",
      "flexDirection": "y",
      "flexAlign": "start",
      "flexJustify": "start",
      "rowGap": 20,
      "columnGap": 20,
      "paddingTop": 19,
      "paddingRight": 19,
      "paddingBottom": 19,
      "paddingLeft": 19
    },
    "1BwWRgjHyEzJjsehRefVBE": {
      "hidden": false,
      "width": { "type": "hug" },
      "height": { "type": "hug" },
      "topLeftRadius": 8,
      "topRightRadius": 8,
      "bottomRightRadius": 8,
      "bottomLeftRadius": 8,
      "fills": [{ "type": "solid", "color": "#f0f9ff" }],
      "opacity": 1,
      "overflowHidden": false,
      "layout": "flex",
      "flexDirection": "x",
      "flexAlign": "start",
      "flexJustify": "start",
      "rowGap": 10,
      "columnGap": 10,
      "paddingTop": 3,
      "paddingRight": 3,
      "paddingBottom": 3,
      "paddingLeft": 3
    },
    "1BwWRgjHr8HNgX8BRooGv7": {
      "width": { "type": "hug" },
      "height": { "type": "hug" },
      "svgContent": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\" fill=\"none\" xmlns=\"http://www.w3.org/2000/svg\">\n<path d=\"M17.25 6.75L22.5 12L17.25 17.25M6.75 17.25L1.5 12L6.75 6.75M14.25 3.75L9.75 20.25\" stroke=\"#0EA5E9\" stroke-width=\"1.5\" stroke-linecap=\"round\" stroke-linejoin=\"round\"/>\n</svg>\n"
    },
    "1BwWRgjHmYZ8qLKDj3Tvdy": {
      "hidden": false,
      "width": { "type": "fill", "value": 315.640625 },
      "height": { "type": "hug" },
      "fills": [{ "type": "solid", "color": "#1f2937" }],
      "opacity": 1,
      "textContent": "React Components Import/Export",
      "fontFamily": "Satoshi",
      "fontWeight": 700,
      "fontSize": 20,
      "letterSpacing": 0,
      "textHorizontalAlign": "start",
      "textVerticalAlign": "start"
    },
    "1BwWRgjHuAe9UTxVhowwYu": {
      "hidden": false,
      "width": { "type": "fill" },
      "height": { "type": "hug" },
      "fills": [{ "type": "solid", "color": "#4b5563" }],
      "opacity": 1,
      "textContent": "Seamlessly work with React components, boosting your design-to-development workflow.\n",
      "fontFamily": "Inter",
      "fontWeight": 400,
      "fontSize": 16,
      "letterSpacing": 0,
      "textHorizontalAlign": "start",
      "textVerticalAlign": "start"
    },
    "1BwspVKcGYvxVctYVzjPu6": {
      "hidden": true,
      "width": { "type": "fixed", "value": 40 },
      "height": { "type": "fixed", "value": 80 },
      "fills": [{ "type": "solid", "color": "#22C55E" }]
    },
    "1Bvi39mct9UNYZ9eTCCRdV:1BwspVKcGYvxVctYVzjPu6": { "hidden": false }
  }
}
