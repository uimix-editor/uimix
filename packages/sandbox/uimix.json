{
  "prebuiltAssets": ["@uimix/mui-assets"],
  "assets": "./src/assets.ts"
}
