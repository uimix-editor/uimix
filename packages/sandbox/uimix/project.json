{
  "componentURLs": [
    "https://cdn.jsdelivr.net/gh/uimix-editor/uimix@ba0157d5/packages/sandbox/dist-components/components.js",
    "https://cdn.jsdelivr.net/gh/uimix-editor/uimix@ba0157d5/packages/sandbox/dist-components/style.css"
  ]
}
