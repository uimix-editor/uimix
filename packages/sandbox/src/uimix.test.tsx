import reactRenderer from "react-test-renderer";
import { describe, expect, it } from "vitest";
import { Card, Page } from "./components.uimix.js";
import { Button } from "./button.uimix.js";

describe(Button.name, () => {
  it("render", async () => {
    expect(
      reactRenderer
        .create(
          <Button
            className="other-class"
            textProps={{
              children: "Hello, world!",
            }}
          />
        )
        .toJSON()
    ).toMatchSnapshot();
  });
});

describe(Page.name, () => {
  it("render", async () => {
    expect(
      reactRenderer
        .create(
          <Page
            buttonProps={{
              label: "Override Button Label",
            }}
            stackProps={{
              // TODO: typing
              // @ts-ignore
              textProps: {
                children: "Override inner item",
              },
            }}
          />
        )
        .toJSON()
    ).toMatchSnapshot();
  });
});

describe(Card.name, () => {
  it("render", async () => {
    expect(reactRenderer.create(<Card />).toJSON()).toMatchSnapshot();
  });
});
