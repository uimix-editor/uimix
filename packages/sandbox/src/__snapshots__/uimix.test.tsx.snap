// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`render 1`] = `
<div
  className="uimix-Buttonv4qeOq other-class"
>
  <div
    className="uimix-Buttonv4qeOq__text"
  >
    Hello, world!
  </div>
</div>
`;

exports[`render 2`] = `
<div
  className="uimix-PageRBRbRk"
>
  <header
    className="uimix-PageRBRbRk__header"
  >
    <div
      className="wrapper"
    >
      <div>
        <svg
          height="32"
          viewBox="0 0 32 32"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fillRule="evenodd"
          >
            <path
              d="M10 0h12a10 10 0 0110 10v12a10 10 0 01-10 10H10A10 10 0 010 22V10A10 10 0 0110 0z"
              fill="#FFF"
            />
            <path
              d="M5.3 10.6l10.4 6v11.1l-10.4-6v-11zm11.4-6.2l9.7 5.5-9.7 5.6V4.4z"
              fill="#555AB9"
            />
            <path
              d="M27.2 10.6v11.2l-10.5 6V16.5l10.5-6zM15.7 4.4v11L6 10l9.7-5.5z"
              fill="#91BAF8"
            />
          </g>
        </svg>
        <h1>
          Acme
        </h1>
      </div>
      <div>
        <button
          className="storybook-button storybook-button--small storybook-button--secondary "
          style={
            {
              "backgroundColor": undefined,
            }
          }
          type="button"
        >
          Log in
        </button>
        <button
          className="storybook-button storybook-button--small storybook-button--primary "
          style={
            {
              "backgroundColor": undefined,
            }
          }
          type="button"
        >
          Sign up
        </button>
      </div>
    </div>
  </header>
  <div
    className="uimix-PageRBRbRk__frame"
  >
    <button
      className="storybook-button storybook-button--large storybook-button--primary uimix-PageRBRbRk__button"
      style={
        {
          "backgroundColor": "red",
        }
      }
      type="button"
    >
      Override Button Label
    </button>
    <div
      className="uimix-PageRBRbRk__instance uimix-Buttonv4qeOq"
    >
      <div
        className="uimix-PageRBRbRk__instance-Buttonv4qeOq__text uimix-Buttonv4qeOq__text"
      >
        Override Name
      </div>
    </div>
    <img
      alt="Image"
      className="uimix-PageRBRbRk__image"
      src="/src/images/icon.png"
    />
    <div
      className="uimix-PageRBRbRk__stack uimix-StackRBRbRk"
    >
      <div
        className="uimix-PageRBRbRk__stack-StackRBRbRk__item_0 uimix-StackRBRbRk__item_0"
      />
      <div
        className="uimix-PageRBRbRk__stack-StackRBRbRk__item_1 uimix-StackRBRbRk__item_1"
      />
      <div
        className="uimix-PageRBRbRk__stack-StackRBRbRk__hidden_Item uimix-StackRBRbRk__hidden_Item"
      />
      <div
        className="uimix-PageRBRbRk__stack-StackRBRbRk__item_2 uimix-StackRBRbRk__item_2"
      />
      <div
        className="uimix-PageRBRbRk__stack-StackRBRbRk__text uimix-StackRBRbRk__text"
      >
        Override inner item
      </div>
      <div
        className="uimix-PageRBRbRk__stack-StackRBRbRk__button uimix-StackRBRbRk__button"
      >
        <div
          className="uimix-PageRBRbRk__stack-StackRBRbRk__button-Buttonv4qeOq__text uimix-StackRBRbRk__button-Buttonv4qeOq__text uimix-Buttonv4qeOq__text"
        >
          Nested Override
        </div>
      </div>
    </div>
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-1yu95zv-MuiButton-root{font-family:\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif;font-weight:500;font-size:0.875rem;line-height:1.75;letter-spacing:0.02857em;text-transform:uppercase;min-width:64px;padding:6px 16px;border-radius:4px;-webkit-transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;color:#fff;background-color:#1976d2;box-shadow:0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);}.css-1yu95zv-MuiButton-root:hover{-webkit-text-decoration:none;text-decoration:none;background-color:#1565c0;box-shadow:0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);}@media (hover: none){.css-1yu95zv-MuiButton-root:hover{background-color:#1976d2;}}.css-1yu95zv-MuiButton-root:active{box-shadow:0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);}.css-1yu95zv-MuiButton-root.Mui-focusVisible{box-shadow:0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);}.css-1yu95zv-MuiButton-root.Mui-disabled{color:rgba(0, 0, 0, 0.26);box-shadow:none;background-color:rgba(0, 0, 0, 0.12);}",
        }
      }
      data-emotion="css 1yu95zv-MuiButton-root"
    />
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-sghohy-MuiButtonBase-root-MuiButton-root{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;position:relative;box-sizing:border-box;-webkit-tap-highlight-color:transparent;background-color:transparent;outline:0;border:0;margin:0;border-radius:0;padding:0;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;vertical-align:middle;-moz-appearance:none;-webkit-appearance:none;-webkit-text-decoration:none;text-decoration:none;color:inherit;font-family:\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif;font-weight:500;font-size:0.875rem;line-height:1.75;letter-spacing:0.02857em;text-transform:uppercase;min-width:64px;padding:6px 16px;border-radius:4px;-webkit-transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;color:#fff;background-color:#1976d2;box-shadow:0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);}.css-sghohy-MuiButtonBase-root-MuiButton-root::-moz-focus-inner{border-style:none;}.css-sghohy-MuiButtonBase-root-MuiButton-root.Mui-disabled{pointer-events:none;cursor:default;}@media print{.css-sghohy-MuiButtonBase-root-MuiButton-root{-webkit-print-color-adjust:exact;color-adjust:exact;}}.css-sghohy-MuiButtonBase-root-MuiButton-root:hover{-webkit-text-decoration:none;text-decoration:none;background-color:#1565c0;box-shadow:0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);}@media (hover: none){.css-sghohy-MuiButtonBase-root-MuiButton-root:hover{background-color:#1976d2;}}.css-sghohy-MuiButtonBase-root-MuiButton-root:active{box-shadow:0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);}.css-sghohy-MuiButtonBase-root-MuiButton-root.Mui-focusVisible{box-shadow:0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);}.css-sghohy-MuiButtonBase-root-MuiButton-root.Mui-disabled{color:rgba(0, 0, 0, 0.26);box-shadow:none;background-color:rgba(0, 0, 0, 0.12);}",
        }
      }
      data-emotion="css sghohy-MuiButtonBase-root-MuiButton-root"
    />
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium uimix-PageRBRbRk__mUIButton css-sghohy-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      label="Button"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      MUI Button
    </button>
  </div>
</div>
`;

exports[`render 3`] = `
<div
  className="uimix-CardRBRbRk"
>
  <div
    className="uimix-CardRBRbRk__icon"
  >
    <svg
      className="uimix-CardRBRbRk__heroicons_code_bracket"
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M17.25 6.75L22.5 12L17.25 17.25M6.75 17.25L1.5 12L6.75 6.75M14.25 3.75L9.75 20.25"
        stroke="#0EA5E9"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="1.5"
      />
    </svg>
  </div>
  <div
    className="uimix-CardRBRbRk__title"
  >
    React Components Import/Export
  </div>
  <div
    className="uimix-CardRBRbRk__description"
  >
    Seamlessly work with React components, boosting your design-to-development workflow.

  </div>
</div>
`;
